packages/dev/core/src/Maths/ThinMaths/thinMath.matrix.functions.ts

```diff
@@ -9,7 +9,7 @@ export class MatrixManagement {
 }
 
 function SetMatrixData(
-    result: ThinMatrix,
+    result: IMatrixLike,

```

Comment:
An alternative to removing `DeepImmutable` from `IMatrixLike`'s `toArray()` would be to cast it away as needed, like here in this function (or below in places like `InvertMatrixToRef`, etc).